### DISCUSSION TITLE
Watch With SetValue Causing looped behaviour

### DISCUSSION THREAD

**Discussion by MJUSER** [react-hook-form/react-hook-form#5743] [Q&A] [answered]
I have one page which is having a quantity field. So on change of quantity, I need to calculate something and store it to redux.  This design repeats on a tab , as user can add as many tabs dynamically with this same form structure. So when am updating quantity i am using watch > then calls a function to update redux. After the update of redux changes am subscribing to the tab, though setting values back to quantity again. Here am using setValue('quantity) to do the same. Which again causes on change and calling watch > makes a loop of watch() and setValue()

use watch .. same value we need to update back to input through setValue()

not including a code sandbox
react hook form v6 using, windows 10 OS, browser chrome.

**Comment by JGibel** [comment, marked as answer]
If the quantity isn't actually being updated, do a comparison check before calling setValue. 

You should include a CSB, but it doesn't sound like a RHF issue.
